### PR TITLE
Feature/delete

### PR DESCRIPTION
**Pull request by Shahar-Y**
Added delete feature to files. Also tested it.

### CODE REVIEW

File: src/file/file.service.ts [hunk 1]
```diff
@@ -106,13 +107,16 @@ export class FileService {
     return file;
   }
 
-  public static async getFilesByFolder(folderID: string | null, ownerID: string | null): Promise<any> {
+  // isDeleted chooses if it would send back the deleted files or not. by default retrieves non-deleted.
+  public static async getFilesByFolder(folderID: string | null, ownerID: string | null, retrieveDeleted?: boolean): Promise<any> {
```
Comment (AllKinds) [inline src/file/file.service.ts]: define isDeleted default value to be false

File: src/file/file.service.ts [hunk 2]
```diff
@@ -78,16 +78,24 @@ export class FileService {
...
     });
 
     return await FilesRepository.create(file);
   }
 
+  // recursive
   public static async delete(fileId: string): Promise<void> {
     // TODO: Delete all children
+    const file: IFile = await this.getById(fileId);
+    if (file.type === 'Folder') {
```
Comment (omrishtam) [inline src/file/file.service.ts]: `file.type === 'Folder'` wtf? Why isn't it a constant and why is it named like this?
Also remove // TODO comment...

File: src/file/file.service.ts [hunk 3]
```diff
@@ -106,13 +114,15 @@ export class FileService {
     return file;
   }
 
-  public static async getFilesByFolder(folderID: string | null, ownerID: string | null): Promise<any> {
+  // isDeleted chooses if it would send back the deleted files or not. by default retrieves non-deleted.
```
Comment (omrishtam) [inline src/file/file.service.ts]: This comment is still not helpful.

Comment (Shahar-Y) [reply src/file/file.service.ts]: why? it is not clear from the name. How would you recommend to explain it?

Comment (omrishtam) [reply src/file/file.service.ts]: This comment only explains what `isDeleted` is used for, but doesn't explain what the function does.

Comment (Shahar-Y) [reply src/file/file.service.ts]: So it doesn't mean its not helpful. Its true that functions should have more comments. Deleting helpful comments won't fix it.

Comment (omrishtam) [reply src/file/file.service.ts]: Just comment the function :angry:

**Pull request merged**